pkg/synthetic/ticket.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package synthetic generates fake data for testing.
+package synthetic
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"open-match.dev/open-match/pkg/pb"
+	"open-match.dev/open-match/pkg/structs"
+)
+
+// Tickets generates artificial tickets with these properties:
+// mmr.rating = value 0 to 100, average 50, standard deviation of 15.
+// one of the role properties:
+//   role.dps = 1 with 40% chance
+//   role.support = 1 with 33% chance
+//   role.tank = 1 with 27% chance
+// mode.demo = 1 to 5 (inclusive) with decreasing percentage each mode.
```

Comment:
how about mode.demo, mode.ctf, mode.battleroyale and such being the modes - and the one that the user picks is on the ticket?

Reply:
I wanted different types of input variations to be represented.  The roles already do what you mention.